File: src/hello_world.js
/**
 * My First JavaScript Program
 * A simple Hello World program to get started with JavaScript.
 */

/**
 * Greet someone by name.
 * @param {string} name - The name to greet. Defaults to "World".
 * @returns {string} A greeting message.
 */
function greet(name = "World") {
    return `Hello, ${name}!`;
}

/**
 * Main function to run the program.
 */
function main() {
    console.log(greet());
    console.log(greet("GitHub"));
    console.log(greet("Programmer"));
    
    // Try it with your own name!
    console.log("\nTry customizing the greeting:");
    console.log(greet("Your Name Here"));
}

// Run the program
main();

File: src/hello_world.py
#!/usr/bin/env python3
"""
My First Python Program
A simple Hello World program to get started with programming.
"""

def greet(name="World"):
    """
    Greet someone by name.
    
    Args:
        name (str): The name to greet. Defaults to "World".
    
    Returns:
        str: A greeting message.
    """
    return f"Hello, {name}!"


def main():
    """
    Main function to run the program.
    """
    print(greet())
    print(greet("GitHub"))
    print(greet("Programmer"))
    
    # Try it with your own name!
    print("\nTry customizing the greeting:")
    print(greet("Your Name Here"))


if __name__ == "__main__":
    main()
